mautic/mautic
# Added warning if the mailer is the owner is different from global setting

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/317
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | ?
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR is based on https://github.com/mautic/mautic/pull/6071.

⚠️ when rebasing this PR from 2019 to Mautic 5 I found that most of the code is already here. So this PR is basically just adding this new warning:

![Screenshot 2024-02-07 at 12 32 34](https://github.com/mautic/mautic/assets/1235442/d3c9aca5-2346-4ca2-8d3f-465c7bf86488)

The rest including the new Email property was there so I removed the migration.

### Steps to test:

1. Create a new email
2. Check that if you use different setting for mailer is owner than the default value from the global config then it shows the warning.

## The original description:

There is new switch in each email form to decide whether the email should use the mailer is owner feature. When creating a new email it will be pre-set to whatever is set in the global configuration. Once an email is saved it will use that option on sending.

A migration exists that will add this new option to all existing emails and set it to whatever you currently have configured in the configuration to keep BC.

A warning will be displayed if the local Mailer is Owner option overwrites the global option:
![Screenshot 2019-07-29 at 15 56 27](https://user-images.githubusercontent.com/1235442/62054273-8a8e0300-b219-11e9-81cb-38ec9b3de4ad.png)


#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
3. Create a segment with at least 2 contact. One contact will have an owner, the other won't.
4. Ensure that your user has different email address that Mautic default from address.
5. Create a segment email called *Mailer is owner turned OFF* and turn off the new switch.
6. Send the email and check that both emails were delivered from the email address in the global configuration.
7. Create a segment email called *Mailer is owner turned ON* and turn the switch on.
8. Send the email and check that one email was sent from your user's email and the other form the email address in the configuration.
9. Create a segment email called *Mailer is owner turned OFF with from set* and turn off the new switch and set a from address for the email.
10. Send the email and check that both emails were delivered from the email address in the email from field.
11. Create a segment email called *Mailer is owner turned ON with from set* and turn the switch on and set a from address for the email.
12. Send the email and check that one email was sent from your user's email and the other form the email address in the email from field.
13. Test with queued emails as well as with immediate sends.
14. Test with campaign emails too.
